contracts/test/EthStorageContractTest.t.sol

```diff
@@ -99,4 +99,27 @@ contract EthStorageContractTest is Test {
         assertEq(storageContract.size(bytes32(uint256(1))), 20);
         assertEq(storageContract.size(bytes32(uint256(2))), 30);
     }
+
+    function testPutBlobsXshard() public {
+        uint256 size = 6;
+        bytes32[] memory keys = new bytes32[](size);
+        uint256[] memory blobIdxs = new uint256[](size);
+        uint256[] memory lengths = new uint256[](size);
+        for (uint256 i = 0; i < size; i++) {
+            keys[i] = bytes32(uint256(i));
+            blobIdxs[i] = i;
+            lengths[i] = 10 + i * 10;
+        }
+
+        uint256 sufficientCost = storageContract.upfrontPaymentInBatch(size);
```

Comment:
Does this test cover the case of opening a new shard?

Reply:
A shard in the test has 4 entries each, and the batch size is 6, so yes.